src/ResourceManager/Batch/Commands.Batch/WorkItems/GetBatchWorkItemCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Batch;
+using Microsoft.Azure.Commands.Batch.Models;
+using Microsoft.Azure.Commands.Batch.Properties;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Constants = Microsoft.Azure.Commands.Batch.Utils.Constants;
+
+namespace Microsoft.Azure.Commands.Batch
```

Comment:
Hmm, I'd prefer to call WriteObject(collection, false) in the case that I'm actually returning a collection.  We have a custom enumerator that has some features that let it handle large scale data sets pretty well, such as only querying the service in chunks when the enumerator is actually being iterated through and supporting parallel iteration very well.  Walking through the entire collection up front removes some of these benefits.data into memory unnecessarily.  Would this level of logic in the cmdlet be ok?
if (name not null) { WriteObject(client.GetSingleItem()); }
else { WriteObject(client.GetCollection(), false); }